SvmTask.py

```diff
@@ -0,0 +1,565 @@
+    df_labels = read_from_cassandra(sqlContext, my_keyspace, labels_table_name)
+   
+    feat_set_id = df_feats.select(df_feats['feature_set_id']).take(1)[0].feature_set_id
+    print 'Using feature set id : '
+    print feat_set_id
+   
+    # udf_label = udf(getClassLabel, IntegerType())
+    # dx = df_labels.withColumn('class_labels', udf_label(df_labels['labels']))
+    #dx.show()//
+    res_df = df_feats.filter(df_feats['feature_set_id'] == feat_set_id).join(df_labels, df_feats.image_id == df_labels.image_id).select(df_feats['image_id'], df_feats['feature_vector'], df_labels['labels'])
+    #res_df_labeled = res_df.withColumn('classLabel', getClassLabel(res_df.labels))
+    exprs = [col("labels").getItem(k).alias(k) for k in keys]
+    trans_data = res_df.select(res_df.image_id,res_df.feature_vector, *exprs)
+    trans_data = trans_data.na.fill(0)
+   
+    return trans_data
+ 
+
+
+# In[39]:
+
+def createResTable(results_table_name):
+    quick_cql("CREATE TABLE IF NOT EXISTS \"%s\".\"%s\"( query_id uuid, score float, image_id uuid, PRIMARY KEY ((query_id), score) ) WITH caching = { 'keys' : 'ALL', 'rows_per_partition' : '100000' };"% (my_keyspace, results_table_name)
+    )
+    results_schema = StructType([
+        StructField('query_id', StringType(), True),
+        StructField('image_id', StringType(), True),
+        StructField('score', FloatType(), True)
+    ])
+    return results_schema
+
+
+# In[48]:
+
+
+from pyspark.ml import Pipeline
+from pyspark.ml.evaluation import BinaryClassificationEvaluator
+from pyspark.ml.tuning import CrossValidator, ParamGridBuilder
+from pyspark.sql import Row
+from pyspark.sql import functions as F
+from pyspark.mllib.classification import SVMWithSGD, SVMModel
+from pyspark.mllib.regression import LabeledPoint
+from pyspark.sql.functions import col
+
+#Paramters to tune
+my_iterations = [5000];
+my_regParam = [0.01, 0.001]
+my_step = [ 2**j for j in range(-3, -2) ]
+numFolds=3
+
+print 'prepare data'
+trans_data = prepareData()
+labeled_df = trans_data.where((trans_data.cat == 1) | (trans_data.dog == 1) | (trans_data.hamster == 1)).withColumn('label', F.when( (trans_data.cat == 1.0), 1.0).otherwise(0.0))
+
+#Randomly shuffle data
+#labeled_df.orderBy(rand())
+test_df = trans_data.where((trans_data.cat != 1) & (trans_data.dog != 1) & (trans_data.hamster != 1))
+
+
+val_range = [ j for j in range(0, numFolds) ]
+split_range =  [ 1.0/numFolds for j in range(0, numFolds)]
+gl_valid_err = 2
+best_i = 0
+best_j = 0
+best_k = 0
+
+split_data = labeled_df.randomSplit(split_range)
+split_train_data = []
+split_val_data = []
+for t in val_range:
+    split_val_data.insert(t, split_data[t])
+    for v in val_range:
+        if t!=v:
+            if len(split_train_data) > t:
+                split_train_data[t] = split_train_data[t].unionAll(split_data[v])
+            else:
+                split_train_data.insert(t, split_data[v])
+
```

Comment:
Seems like the train_val split could be simplified significantly with appropriate indexing rather than loops